src/pages-and-resources/discussions/app-list/AppList.test.jsx

```diff
@@ -1,21 +1,124 @@
+import React from 'react';
+import { initializeMockApp } from '@edx/frontend-platform';
+import MockAdapter from 'axios-mock-adapter';
+import {
+  queryByText, render, queryAllByRole, queryByRole, getByRole, queryByLabelText,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { AppProvider } from '@edx/frontend-platform/react';
+import { Context as ResponsiveContext } from 'react-responsive';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import AppList from './AppList';
+import initializeStore from '../../../store';
+import { fetchApps } from '../data/thunks';
+import { executeThunk } from '../data/redux.test';
+import { getAppsUrl } from '../data/api';
+import { legacyApiResponse, piazzaApiResponse } from '../factories/mockApiResponses';
+import { updateStatus, LOADED, selectApp } from '../data/slice';
+
+const courseId = 'course-v1:edX+TestX+Test_Course';
+
 describe('AppList', () => {
-  test('displays a message when there are no apps available', () => {
+  let axiosMock;
+  let store;
+  let container;
+
+  function createComponent(screenWidth = 1280) {
+    return (
+      <AppProvider store={store}>
+        <ResponsiveContext.Provider value={{ width: screenWidth }}>
+          <IntlProvider locale="en" messages={{}}>
+            <AppList />
+          </IntlProvider>
+        </ResponsiveContext.Provider>
+      </AppProvider>
+    );
+  }
 
+  beforeEach(async () => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: true,
+        roles: [],
+      },
+    });
+
+    store = await initializeStore();
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
   });
 
-  test('display a card for each available app', () => {
+  afterEach(() => {
 
   });
 
-  test('displays the FeaturesTable at desktop sizes', () => {
+  test('displays a message when there are no apps available', async () => {
+    // selects an app to mock store data
+    store.dispatch(selectApp({ appId: 'legacy' }));
+    // update store status to LOADED
+    store.dispatch(updateStatus({ status: LOADED }));
+    // renders component
+    const component = createComponent();
+    const wrapper = render(component);
+    container = wrapper.container;
+    // assert test case results
+    expect(queryByText(container, 'There are no discussions providers available for your course.')).toBeInTheDocument();
```

Comment:
rather than hardcoding this string here, can we fetch it from the `messages.js`?